src/Microsoft.Health.Dicom.Core/Features/Query/IQueryParser.cs

```diff
@@ -3,13 +3,15 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
+using System.Collections.Generic;
+using Dicom;
 using Microsoft.Health.Dicom.Core.Features.Query.Model;
 using Microsoft.Health.Dicom.Core.Messages.Query;
 
 namespace Microsoft.Health.Dicom.Core.Features.Query
 {
     public interface IQueryParser
     {
-        QueryExpression Parse(QueryResourceRequest request);
+        QueryExpression Parse(QueryResourceRequest request, IDictionary<QueryResource, HashSet<DicomTag>> customTagMapping = null);
```

Comment:
should customtag status also  be passed through? Or just filter out Reindexing and Deindexing custom tags before calling this?